core/src/main/java/org/apache/iceberg/ManifestFilterManager.java

```diff
@@ -443,38 +443,36 @@ private ManifestFile filterManifestWithDeletedFiles(
-                  if (entry.status() != ManifestEntry.Status.DELETED) {
-                    if (markedForDelete || evaluator.rowsMightMatch(file)) {
-                      boolean allRowsMatch = markedForDelete || evaluator.rowsMustMatch(file);
-                      ValidationException.check(
-                          allRowsMatch
-                              || isDelete, // ignore delete files where some records may not match
-                          // the expression
-                          "Cannot delete file where some, but not all, rows match filter %s: %s",
-                          this.deleteExpression,
-                          file.location());
-
-                      if (allRowsMatch) {
-                        writer.delete(entry);
-
-                        if (deletedFiles.contains(file)) {
-                          LOG.warn(
-                              "Deleting a duplicate path from manifest {}: {}",
-                              manifest.path(),
-                              file.location());
-                          duplicateDeleteCount += 1;
-                        } else {
-                          // only add the file to deletes if it is a new delete
-                          // this keeps the snapshot summary accurate for non-duplicate data
-                          deletedFiles.add(file.copyWithoutStats());
-                        }
+                  if (markedForDelete || evaluator.rowsMightMatch(file)) {
+                    boolean allRowsMatch = markedForDelete || evaluator.rowsMustMatch(file);
+                    ValidationException.check(
+                        allRowsMatch
+                            || isDelete, // ignore delete files where some records may not match
+                        // the expression
+                        "Cannot delete file where some, but not all, rows match filter %s: %s",
+                        this.deleteExpression,
+                        file.location());
+
+                    if (allRowsMatch) {
+                      writer.delete(entry);
+
+                      if (deletedFiles.contains(file)) {
+                        LOG.warn(
+                            "Deleting a duplicate path from manifest {}: {}",
+                            manifest.path(),
+                            file.location());
+                        duplicateDeleteCount += 1;
                       } else {
-                        writer.existing(entry);
+                        // only add the file to deletes if it is a new delete
+                        // this keeps the snapshot summary accurate for non-duplicate data
+                        deletedFiles.add(file.copyWithoutStats());
                       }
-
                     } else {
                       writer.existing(entry);
                     }
+
+                  } else {
+                    writer.existing(entry);
                   }
                 });
       } finally {
```

Comment:
We could also use try-with-resources here, but I don't want to clutter up the diff.